rectools/model_selection/kfold_split.py

```diff
@@ -124,12 +124,9 @@ def _split_without_filter(
         shuffled_idx = rng.permutation(idx)
         for i in range(self.n_splits):
             fold_info = {"fold_number": i}
-            test_mask = np.zeros_like(idx, dtype=bool)
-            chosen_idx = shuffled_idx[i * test_part_size : (i + 1) * test_part_size]
-            test_mask[chosen_idx] = True
-            train_mask = ~test_mask
-
+            train_mask = np.ones_like(idx, dtype=bool)
+            test_idx = shuffled_idx[i * test_part_size : (i + 1) * test_part_size]
+            train_mask[test_idx] = False
             train_idx = idx[train_mask].values
```

Comment:
We can take it a step further

```suggestion
            left = i * test_part_size
            right = (i + 1) * test_part_size
            test_idx = shuffled_idx[left : right]
            train_idx = np.concatenate((shuffled_idx[:left], shuffled_idx[right:]))
```

It's a bit simpler and approx twice faster. But `train_idx` will be shuffled

Reply:
Done